rapids-cmake/cython-core/create_modules.cmake

```diff
@@ -117,9 +118,10 @@ function(rapids_cython_create_modules)
     add_custom_command(OUTPUT ${cpp_filename}
                        DEPENDS ${cython_filename}
                        VERBATIM
-                       COMMAND "${CYTHON}" ARGS "${_language_flag}" -3 ${CYTHON_FLAGS_LIST}
```

Comment:
Note this change is technically unrelated to the dependency file feature, however, without it I would get:
```sh
[1/3] Transpiling test.pyx to test.c
FAILED: test.c /Users/jfaibussowit/soft/nv/rapids-cmake/testing/cython-core/cython-core_create_modules/build/test.c
.. && /Users/jfaibussowit/soft/packages/base_venv/bin/cython "" -3 --directive ...
/Users/jfaibussowit/soft/packages/base_venv/bin/cython: No such file or directory: ''
```
Note the `""` argument where `${_language_flag}` should be since:
```cmake
set(_ext ".c")
set(_language_flag "")
if(_RAPIDS_CYTHON_CXX) # from function arguments
  set(_language_flag "--cplus")
  set(_ext ".cxx")
endif()
```
Not sure how CI did not catch this error previously @robertmaynard?

Reply:
presumably `_RAPIDS_CYTHON_CXX` is always set in all of our tests, so we never ran into the error.

Reply:
`_RAPIDS_CYTHON_CXX` is due to the preceding `cmake_parse_arguments()` call. The test does not pass the `CXX` argument, so `_RAPIDS_CYTHON_CXX` is always false.

It might be that this test is never run?

Reply:
I think that you're both partially right. The tests are being run in CI, but [these tests are all added with `add_cmake_config_test`](https://github.com/rapidsai/rapids-cmake/blob/branch-24.06/testing/cython-core/CMakeLists.txt#L22), which means only the CMake configure happens, not the build. We can change those to `add_cmake_build_test`. I can reproduce the error you observe when I add that, and your update fixes that. In practice all of RAPIDS passes `CXX` so we don't actually ever use this API without the flag in practice.

I am not sure how you were observing this failure locally without making that change, but good find!

Reply:
> I am not sure how you were observing this failure locally without making that change, but good find!

I just did a `cmake -S . -B build` then `cmake --build build` inside the directory containing these files :)